app/channels/discussion_post_list_channel.rb

```diff
@@ -0,0 +1,14 @@
+class DiscussionPostListChannel < ApplicationCable::Channel
+  def self.notify!(discussion, iteration)
+    broadcast_to("#{discussion.id}_#{iteration.id}", {})
+  end
+
+  def subscribed
+    discussion = Solution::MentorDiscussion.find(params[:discussion_id])
+    iteration = discussion.solution.iterations.find(params[:iteration_id])
+
+    return unless current_user == discussion.mentor || current_user == discussion.student
```

Comment:
Could you add a method to `MentorDiscussion` for `viewable_by?` which does this please. Add a test, and also add a `TODO` there that any admins should be able to see any discussion.